var/spack/repos/builtin/packages/py-pennylane/package.py

```diff
@@ -12,15 +12,16 @@ class PyPennylane(PythonPackage):
 
     homepage = "https://docs.pennylane.ai/"
     git = "https://github.com/PennyLaneAI/pennylane.git"
-    pypi = "PennyLane/PennyLane-0.28.0.tar.gz"
+    url = "https://github.com/PennyLaneAI/pennylane/archive/refs/tags/v0.29.0.tar.gz"
 
     maintainers("mlxd", "AmintorDusko", "marcodelapierre")
 
     version("master", branch="master")
-    version("0.29.1", sha256="c5d662994b741afa69e4fdadc79a1b75840275138a8b7e0cfc5fd64b66a12eef")
+    version("0.29.1", sha256="6ecfb305a3898347df8c539a89a67e748766941d159dbef9e34864872f13c45c")
```

Comment:
Hey @vincentmr, I don't get why we have a change in the hashes. Do you happen to know?

Reply:
Yes, that's because I changed the source from `pypi` to `github`, as in the other py-penny packages. If you prefer I can revert everything in this file.

Reply:
I will defer to @mlxd  to decide that. I'm good with both, but I only tested the previous way.

Reply:
Since Github will always be our source of truth and where we create releases, let's favour that across the packages until a time where we can move all to PyPI.

Reply:
For most Python packages we prefer PyPI (https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github), but if you need files that aren't included in the PyPI tarball then it's fine to use GitHub.

Reply:
The long-plan is to improve the source distributions on PyPI so we can build with the same confidence as the Github releases. Happy to update to PyPI tags once this work is finished across all the packages.